packages/client/src/v2-events/features/drafts/useDrafts.ts

```diff
@@ -126,12 +128,17 @@ setMutationDefaults(trpcOptionsProxy.event.draft.create, {
       },
       createdAt: new Date().toISOString()
     }
-    setDraftData((drafts) => drafts.concat(optimisticDraft))
+    setDraftData((drafts) =>
+      drafts
+        .filter((draft) => draft.eventId !== optimisticDraft.eventId)
+        .concat(optimisticDraft)
+    )
+    clearPendingDraftCreationRequests(variables.eventId)
     return optimisticDraft
   },
   onSuccess: async () => {
-    await invalidateEventsList()
-    await invalidateDraftsList()
+    await refetchEventsList()
+    await refetchDraftsList()
```

Comment:
We apparently need to be extra careful with what we invalidate vs what we refetch. Invalidating something will mean the browser needs to reload the data and can't use the cached version